### ISSUE TITLE
Fix false positives for font-face in font-family-no-missing-generic-family-keyword

### ISSUE THREAD

**Issue by xPaw** [stylelint/stylelint#3033]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Just updated to 8.3.0, and the new `font-family-no-missing-generic-family-keyword` rule is producing warnings for `font-family` declarations inside of `@font-face`.

As far as I understand it, the fallback should be specified in elements that use the font, but not inside of `@font-face`, am I misunderstanding it?

> Which rule, if any, is this issue related to?

`font-family-no-missing-generic-family-keyword`

> What CSS is needed to reproduce this issue?

e.g.

```css
@font-face {
	font-family: "FontAwesome";
}
```

> What stylelint configuration is needed to reproduce this issue?

Using `stylelint-config-standard`

```yml
font-family-no-missing-generic-family-keyword: true
```

> Which version of stylelint are you using?

8.3.0

> What did you expect to happen?

No warning.

> What actually happened (e.g. what warnings or errors you are getting)?

```shell
2:15  ×  Unexpected missing generic font family                                 font-family-no-missing-generic-family-keyword
```

**Comment by hudochenkov** [comment]
Could you use issue template, please, so we have better picture.

**Comment by jeddy3** [comment]
@xPaw Thanks for updating your issue to use the template. Yep, looks like [we missed this one](https://www.w3.org/TR/css-fonts-3/#font-face-rule). I've added a pending PR.

**Comment by xPaw** [comment]
Side question, should I still be adding a generic font for icon fonts?

**Comment by jeddy3** [comment]
I not 100% sure, I'm afraid. I doubt [the spec](https://www.w3.org/TR/css-fonts-3/#generic-font-families) anticipated them at the time of writing. I guess there's no harm in adding a generic font after an icon font. It also feels like icon fonts were a workaround for the poor support of SVGs in some browsers, which I believe is no longer the case. So, icon fonts might become an artefact of their time.

**Comment by jeddy3** [comment]
Released in `8.3.1`.

I figured I'd run a quick patch release before work this morning as I suspect a lot of users will run into this bug.

**Comment by hudochenkov** [comment]
Good call, @jeddy3! Thank you!

**Comment by SyedShahjahan** [comment]
Adding the value from Variable has fixed my problem You can too try this trick.

```
$font-family-fontawesome: "FontAwesome";

font-family: $font-family-fontawesome;
```

**Comment by dandee** [comment]
> Adding the value from Variable has fixed my problem You can too try this trick.
> 
> ```
> $font-family-fontawesome: "FontAwesome";
> 
> font-family: $font-family-fontawesome;
> ```

That worked for me. Thanks!

**Comment by jeddy3** [comment]
@dandee You can use the [`ignoreFontFamilies` secondary option](https://stylelint.io/user-guide/rules/list/font-family-no-missing-generic-family-keyword/#ignorefontfamilies-regex-regex-string), rather than that trick.

For example:

```json
{
  "rules": {
    "font-family-no-missing-generic-family-keyword": [true, { 
      "ignoreFontFamilies": ["FontAwesome"] }
    ]
  }
}
```

[Demo](https://stylelint.io/demo#N4Igxg9gJgpiBcICGACYKBmEB2AXAtBkgLYCWANgJ7woA6IAYjrgIIDuMAzhMTPQNwoAvrWwgANCE6U8SAB4JwnThPA4MpAOaLgolHRAAnAK7ku9Gruz799LHkIkKlfNgj4yy0tk35NMbBhDUjBHMip8AGsYSjYIQygLFABtXBMYcTQ6axtbEC03QxgmPAYnclJzBBT6EtYObl56AF1hPRtm9pFsIRAhIA).

**Comment by dandee** [comment]
Hey @jeddy3 !

Thanks for the tip! I'm currently developing an app based on the latest VueJS 2.x. The project has been set up and updated by the most recent VueJS CLI tools. I've added there a "global" SCSS stylesheet that includes all common dependencies and customizations. When I customized styles for one of the libraries by adding "font-family: FontAwesome" in that file I've started getting the error message even though I've explicitly included .css file with the font declaration there. So for me this is still unresolved issue.

<img width="2090" alt="Screenshot 2021-09-19 at 00 33 44" src="https://user-images.githubusercontent.com/890495/133910361-e0a4a969-c4e5-45da-9bff-2a57847a068e.png">

<img width="1275" alt="Screenshot 2021-09-19 at 00 44 39" src="https://user-images.githubusercontent.com/890495/133910370-acfd9781-c036-4ef4-ab5c-147433980e51.png">

**Comment by jeddy3** [comment]
You'll need to update [your stylelint configuration object](https://stylelint.io/user-guide/configure) to the [include the example above](https://github.com/stylelint/stylelint/issues/3033#issuecomment-922370942).

I don't know where VueJS CLI tools' puts it, but they'll be a stylelint config somewhere.
